src/families/osmosis/js-synchronization.ts

```diff
@@ -0,0 +1,46 @@
+import { encodeAccountId } from "../../account";
+import {
+  makeSync,
+  makeScanAccounts,
+  GetAccountShape,
+  mergeOps,
+} from "../../bridge/jsHelpers";
+import { getAccount, getOperations } from "./api";
+
+const getAccountShape: GetAccountShape = async (info) => {
+  const { address, initialAccount, currency, derivationMode } = info;
+  const oldOperations = initialAccount?.operations || [];
+
+  const accountId = encodeAccountId({
+    type: "js",
+    version: "2",
+    currencyId: currency.id,
+    xpubOrAddress: address,
+    derivationMode,
+  });
+
+  const { blockHeight, balance } = await getAccount(address);
+
+  // Merge new operations with the previously synced ones
```

Comment:
Not a big fan of this way to get operations, are we sure we will get all the operations ? 
Is there any info we can get from the RPC / API to get the exact number of operations.

Reply:
Unfortunately we have to allow some uncertainty around what gets returned by the indexer, so it's hard/impossible to check against an exact number of entries. The latest update to this file introduces additional safeguards to parse data and gracefully recover if unexpected data is read.